src/wallet/KintoWallet.sol

```diff
@@ -286,22 +287,28 @@ contract KintoWallet is Initializable, BaseAccount, TokenCallbackHandler, IKinto
             }
             return _packValidationData(false, 0, 0);
         }
-        bytes[] memory signatures = new bytes[](owners.length);
-        // Split signature from userOp.signature
-        if (requiredSigners == 1) {
-            signatures[0] = userOp.signature;
-        } else {
-            signatures = ByteSignature.extractSignatures(userOp.signature, requiredSigners);
-        }
-        for (uint256 i = 0; i < owners.length; i++) {
-            if (owners[i] == hash.recover(signatures[i])) {
-                requiredSigners--;
-                if (requiredSigners == 0) {
-                    break;
+
+        // Multiple signers
+        bool[] memory hasSigned = new bool[](owners.length);
+        bytes[] memory signatures = ByteSignature.extractSignatures(userOp.signature, requiredSigners);
+
+        for (uint256 i = 0; i < signatures.length; i++) {
+            address recovered = hash.recover(signatures[i]);
+
+            for (uint256 j = 0; j < owners.length; j++) {
+                if (owners[j] == recovered && !hasSigned[j]) {
+                    hasSigned[j] = true;
+                    requiredSigners--;
+                    break; // once the owner is found
                 }
             }
         }
-        return _packValidationData(requiredSigners != 0, 0, 0);
+
+        if (requiredSigners == 0) {
```

Comment:
Why not keep this? Its smaller and equivalent

return _packValidationData(requiredSigners != 0, 0, 0);

Reply:
yep, i was testing something and left it like that, but agree it is much cleaner the other way! changing it now

Reply:
pushed!